### DISCUSSION TITLE
How to gracefully reload a document via Postmessage/Wopi

### DISCUSSION THREAD

**Discussion by andrieshiemstra** [CollaboraOnline/online#5474] [Q&A] [answered]
Hi, I'm working on integrating Collabora via wopi/postmessage api and after doing some postprocessing on the server I'd like to reload the document.

Currently is use the "Host_VersionRestore" message
```javascript
this.postMessage("Host_VersionRestore", { "Status": "Pre_Restore" });
```

Although this gets the job done quite nicely there seems to be a lot of repainting going on and for a sec you get a "user X left" tooltip and such.

Is there a more 'subtle' way to actively let the editor know there are changes made to the document on the server and it should reload?

**Comment by pedropintosilva** [comment, marked as answer]
Hello @andrieshiemstra and sorry for the delayed response. It seems we really end up not ussing the 'Discussion' here on GH. For these type of questions you might find useful https://forum.collaboraonline.com/  (there are a couple related with SDK there with answers)

More info at https://sdk.collaboraonline.com/docs/postmessage_api.html?highlight=host_versionrestore#id3

But to answer you directly, I think that's it.  Collabora Online saves anything that is unsaved first and only after restores the document and rendering it to the user  (user ends up leaving the other version and then entering the new restored version/view)

So Maybe that tooltip etc could be handled differently as a feature request: https://github.com/CollaboraOnline/online/issues/new?assignees=&labels=enhancement%2C+unconfirmed&template=feature_request.md&title=

**Reply by andrieshiemstra** [reply]
Thanks Pedro, next time i'll ask in the forum.
